site/app/repositories/email/EmailRepository.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace app\repositories\email;
+
+use app\models\EmailStatusModel;
+use Doctrine\ORM\EntityRepository;
+
+class EmailRepository extends EntityRepository {
+    const PAGE_SIZE = 5000;
+    const MAX_SUBJECTS_PER_PAGE = 10;
+
+    public function getEmailsByPage(int $page, $semester = null, $course = null): iterable {
```

Comment:
You are returning an array of iterable objects. Shouldn't return type be array then? Technically an array is iterable but array is the more specific option.